source/grid/tria.cc

```diff
@@ -968,6 +969,35 @@ namespace
     Assert(periodic_face_map.count(cell_face_1) == 0, ExcInternalError());
     periodic_face_map.insert(periodic_faces);
 
+    if (dim == 1)
+      {
+        if (cell_1->has_children())
```

Comment:
I think this part of the patch is not specific to your work, right? Could you submit this as a separate PR?

(If so, it's really difficult to read code of the form
```
  if (dim==1)
    {
      ...
      return;
    }

  something else
```
because it's so easy to miss the `return` statement. It's substantially easier to read it if you write
```
  if (dim==1)
  {
    // 1d is special because X, Y, and Z. So treat this case separately
    ...
  }
  else  // dim > 1
  {
    ...
  }
```

Reply:
@bangerth you are right! I'll create a separate PR. Is there any test I could use as template for a new test?

Reply:
If I understand correctly, you're here adding periodicity in 1d. Did that just not work at all before?

There are numerous tests for periodicity in `bits/`. It may be simple enough to adapt one of them to run in 1d.

Reply:
@bangerth: thanks! I have created a new PR #8423.